build_tools/release_history.sh

```diff
@@ -0,0 +1,96 @@
+#! /usr/bin/env bash
+
+set -e
+
+if [ ! -d unreleased_history ]; then
+  echo "Can't find unreleased_history/ directory"
+  exit 1
+fi
+
+if [ ! "$DRY_RUN" ]; then
+  # Check for uncommitted changes
+  UNCOMMITTED="$(git diff -- unreleased_history/ HISTORY.md)"
+  if [ "$UNCOMMITTED" ]; then
+    echo 'Uncommitted changes to files to be modified. Please commit first to'
+    echo 'ensure a clean revert path. You can always `git commit -a --amend`'
+    echo 'to add more changes to your commit.'
+    exit 2
+  fi
+fi
+
+# Add first part of existing HISTORY file to new version
+awk '{ print } /NOTE/ { exit(0) }' < HISTORY.md > HISTORY.new
+
+# And a blank line separator
+echo >> HISTORY.new
+
+# Add new version header
+awk '/#define ROCKSDB_MAJOR/ { major = $3 }
+     /#define ROCKSDB_MINOR/ { minor = $3 }
+     /#define ROCKSDB_PATCH/ { patch = $3 }
+     END { print "## " major "." minor "." patch " (" strftime("%F") ")" }' < include/rocksdb/version.h >> HISTORY.new
+
+function process_file () {
+  # use awk to correct extra or missing newlines, missing '* ' on first line
+  awk '/./ { if (notfirstline || $1 == "*") print;
+             else print "* " $0;
+             notfirstline=1; }' < $1 >> HISTORY.new
+  # Part of the reason for putting the header in a file in the dir is because
+  # git won't track empty directories, but we want the directories to stick
+  # around even when there are no unreleased entries for the group.
+  if echo "$1" | grep -q 000_HEADER.md; then
```

Comment:
wondering if we could use `.gitkeep` file for git to track the empty directories.

Reply:
Wasn't aware of that. I can do that to simplify some things, but not a big change.